crates/adapters/src/controller/stats.rs

```diff
@@ -208,6 +345,7 @@ impl ControllerStatus {
             global_metrics: GlobalControllerMetrics::new(),
             inputs: ShardedLock::new(BTreeMap::new()),
             outputs: ShardedLock::new(BTreeMap::new()),
+            metrics: Mutex::new(Vec::new()),
```

Comment:
unrelated but I wondered why two others above are `ShardedLock` whereas this is a Mutex